src/components/Faucet.tsx

```diff
@@ -0,0 +1,179 @@
+"use client";
+
+import React, { useState } from 'react';
+import { Card, CardContent } from "@/components/ui/card";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { isAddress } from 'viem';
+
+const Faucet = () => {
+    const [verification, setVerification] = useState(false);
+    const [userAddress, setUserAddress] = useState('');
+    const [isValidAddress, setIsValidAddress] = useState(true);
+    const [txHash, setTxHash] = useState<string | null>(null);
+    const [lastWithdrawalTime, setLastWithdrawalTime] = useState<bigint | null>(null);
+    const [withdrawalDelay, setWithdrawalDelay] = useState<bigint | null>(null);
+
+    const toastId = React.useRef<string | number | null>(null);
+
+    const checkEligibility = async () => {
+        toastId.current = toast.loading("Verifying Address...");
+        const response = await fetch('/api/faucet/verify', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ address: userAddress })
+        });
+        const result = await response.json();
+
+        if (response.ok) {
```

Comment:
This can also be simplified by using return statements

```
if (response.ok) {
 setVerification(true);
            toast.update(toastId.current, {
                render: result.message,
                isLoading: false,
                autoClose: 5000,
                type: "success",
            });
 return true;
}

if (response.status === 400) {
setVerification(false);
            setLastWithdrawalTime(BigInt(result.lastWithdrawalTime));
            setWithdrawalDelay(BigInt(result.withdrawalDelay));
            toast.update(toastId.current, {
                render: result.message,
                isLoading: false,
                autoClose: 5000,
                type: "error",
            });
            
            return false;
}

 setVerification(false);
            setLastWithdrawalTime(null);
            setWithdrawalDelay(null);
            toast.update(toastId.current, {
                render: result.message || "Address is not verified 😢",
                isLoading: false,
                autoClose: 5000,
                type: "error",
            });
        }
        return false